components/service/fretboard/README.md

```diff
@@ -12,6 +12,213 @@ Use gradle to download the library from JCenter:
 implementation "org.mozilla.components:fretboard:{latest-version}
 ```
 
+### Filters
+Fretboard allows you to specify the following filters:
+- Buckets: Every user is in one of 100 buckets (0-99). For every experiment you can set up a min and max value (0 <= min <= max <= 100). The bounds are [min, max).
+
+- appId (regex): The app ID (package name)
+- version (regex): The app version
+- country (regex): country, pulled from the default locale
+- lang (regex): language, pulled from the default locale
+- device (regex): Android device name
+- manufacturer (regex): Android device manufacturer
+- region: custom region, different from the one from the default locale (like a GeoIP, or something similar).
+For this to work you must provide a RegionProvider implementation when creating the Fretboard instance.
+
+### Creating Fretboard instance
+In order to use the library, first you have to create a new Fretboard instance. You do this once per app launch 
+(typically in your Application class onCreate method). You simply have to instantiate the Fretboard class and
+provide the ExperimentStorage and ExperimentSource implementations, and then call the loadExperiments method
+to load experiments from local storage:
+
+```Kotlin
+class SampleApp : Application() {
+    lateinit var fretboard: Fretboard
+
+    override fun onCreate() {
+        fretboard = Fretboard(
+            experimentSource,
+            experimentStorage
+        )
+        fretboard.loadExperiments()
+    }
+}
+```
+
+Additionally, Fretboard allows you to specify a custom RegionProvider in order to specify a custom region,
+different from the one of the current locale (perhaps doing a GeoIP or something like that):
+
+```Kotlin
+val fretboard = Fretboard(
+    experimentSource,
+    experimentStorage,
+    object : RegionProvider {
+        return custom_region
+    }
+)
+```
+
+
+#### Using Kinto as experiment source
+Fretboard includes a default source implementation for a Kinto backend, which you can use like this:
+
+```Kotlin
+val fretboard = Fretboard(
+    KintoExperimentSource(baseUrl, bucketName, collectionName),
+    experimentStorage
+)
+```
+
+It expects the experiments in the following JSON format:
+```json
+{
+    "data":[
+        {
+            "name": "",
+            "match":{
+                "lang":"",
+                "appId":"",
+                "regions":[],
+                "country":"",
+                "version":"",
+                "device":"",
+                "manufacturer":"",
+                "region":""
+            },
+            "buckets": {
+                "min": "0",
+                "max": "100"
+            },
+            "description":"",
+            "id":"",
+            "last_modified":1523549895713
+        }
+    ]
+}
+```
```

Comment:
This is interesting but move the format all the way to the bottom - it's something a user of the library usually does not need to care about.